File: rerun-reports/lane-b/041/20260828-023205/results/app-configuration/data-plane/python/crud/app-configuration-dp-python-crud/python-azure-skills/azure-skill-mcp/generated-code/app_configuration_manager.py
import os
import sys

from azure.appconfiguration import (
    AzureAppConfigurationClient,
    ConfigurationSetting,
    FeatureFlagConfigurationSetting,
)
from azure.core.exceptions import HttpResponseError


CONNECTION_STRING_ENV = "AZURE_APPCONFIG_CONNECTION_STRING"
SETTING_KEY = "app:Settings:FontSize"
SETTING_VALUE = "24"
PRODUCTION_LABEL = "Production"
KEY_FILTER = "app:Settings:*"


def main() -> int:
    connection_string = os.environ.get(CONNECTION_STRING_ENV)
    if not connection_string:
        print(
            f"Set the {CONNECTION_STRING_ENV} environment variable before running.",
            file=sys.stderr,
        )
        return 2

    client = AzureAppConfigurationClient.from_connection_string(connection_string)

    try:
        client.set_configuration_setting(
            ConfigurationSetting(key=SETTING_KEY, value=SETTING_VALUE)
        )
        client.set_configuration_setting(
            ConfigurationSetting(
                key=SETTING_KEY,
                value=SETTING_VALUE,
                label=PRODUCTION_LABEL,
            )
        )

        setting = client.get_configuration_setting(key=SETTING_KEY)
        print(f"{setting.key} = {setting.value}")

        for matching_setting in client.list_configuration_settings(
            key_filter=KEY_FILTER
        ):
            label = matching_setting.label or "<no label>"
            print(
                f"{matching_setting.key} [{label}] = {matching_setting.value}"
            )

        client.set_configuration_setting(
            FeatureFlagConfigurationSetting(
                feature_id="BetaFeature",
                enabled=True,
            )
        )

        client.delete_configuration_setting(key=SETTING_KEY)
    except HttpResponseError as error:
        status_code = error.status_code or "unknown"
        print(
            f"Azure App Configuration request failed "
            f"(status {status_code}): {error.message}",
            file=sys.stderr,
        )
        return 1
    finally:
        client.close()

    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: rerun-reports/03-java/20260829-075759/results/app-configuration/data-plane/java/crud/app-configuration-dp-java-crud/java-azure-skills/azure-skill-mcp/generated-code/src/main/java/com/example/AppConfigurationManager.java
package com.example;

import com.azure.core.exception.HttpResponseException;
import com.azure.data.appconfiguration.ConfigurationClient;
import com.azure.data.appconfiguration.ConfigurationClientBuilder;
import com.azure.data.appconfiguration.models.ConfigurationSetting;
import com.azure.data.appconfiguration.models.FeatureFlagConfigurationSetting;
import com.azure.data.appconfiguration.models.SettingSelector;

public final class AppConfigurationManager {
    private static final String CONNECTION_STRING_ENVIRONMENT_VARIABLE =
        "AZURE_APPCONFIGURATION_CONNECTION_STRING";
    private static final String FONT_SIZE_KEY = "app:Settings:FontSize";
    private static final String PRODUCTION_LABEL = "Production";

    private AppConfigurationManager() {
    }

    public static void main(String[] args) {
        String connectionString = System.getenv(CONNECTION_STRING_ENVIRONMENT_VARIABLE);
        if (connectionString == null || connectionString.isBlank()) {
            System.err.printf(
                "Set the %s environment variable before running the program.%n",
                CONNECTION_STRING_ENVIRONMENT_VARIABLE);
            System.exit(1);
        }

        ConfigurationClient client = new ConfigurationClientBuilder()
            .connectionString(connectionString)
            .buildClient();

        try {
            client.setConfigurationSetting(FONT_SIZE_KEY, null, "24");

            client.setConfigurationSetting(
                new ConfigurationSetting()
                    .setKey(FONT_SIZE_KEY)
                    .setLabel(PRODUCTION_LABEL)
                    .setValue("24"));

            ConfigurationSetting setting =
                client.getConfigurationSetting(FONT_SIZE_KEY, null);
            System.out.printf("%s = %s%n", setting.getKey(), setting.getValue());

            System.out.println("Matching settings:");
            client.listConfigurationSettings(
                    new SettingSelector().setKeyFilter("app:Settings:*"))
                .forEach(item -> System.out.printf(
                    "  key=%s, label=%s, value=%s%n",
                    item.getKey(),
                    item.getLabel(),
                    item.getValue()));

            FeatureFlagConfigurationSetting betaFeature =
                new FeatureFlagConfigurationSetting("BetaFeature", true);
            client.setConfigurationSetting(betaFeature);

            client.deleteConfigurationSetting(FONT_SIZE_KEY, null);
            System.out.printf("Deleted the unlabeled setting %s.%n", FONT_SIZE_KEY);
        } catch (HttpResponseException exception) {
            int statusCode = exception.getResponse() == null
                ? -1
                : exception.getResponse().getStatusCode();
            System.err.printf(
                "Azure App Configuration request failed (HTTP %d): %s%n",
                statusCode,
                exception.getMessage());
            System.exit(1);
        }
    }
}
